## ACTIVIDAD OBTENCIÓN DE ESTADÍSTICA DESCRIPTIVAS

Fátima Sofía Sacoto Idrovo  
IDM  
A00841256

In [5]:
import pandas as pd
import numpy as np

In [2]:
diabetes = pd.read_csv("diabetes.csv")

In [6]:
np.shape(diabetes)

(768, 9)

In [7]:
diabetes.head(5)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [8]:
diabetes.tail(5)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [9]:
diabetes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [10]:
diabetes.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                51
Insulin                     186
BMI                         248
DiabetesPedigreeFunction    517
Age                          52
Outcome                       2
dtype: int64

In [11]:
diabetes.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [16]:
mean_insulin = diabetes['Insulin'].mean()
median_insulin = diabetes['Insulin'].median()
mode_insulin = diabetes['Insulin'].mode()
print("Mean_insulin:",mean_insulin)
print("Median_insulin:",median_insulin)
print("Mode_insulin:",mode_insulin)

Mean_insulin: 79.79947916666667
Median_insulin: 30.5
Mode_insulin: 0    0
Name: Insulin, dtype: int64


In [17]:
mean_BMI = diabetes['BMI'].mean()
median_BMI = diabetes['BMI'].median()
mode_BMI = diabetes['BMI'].mode()
print("Mean_BMI:",mean_BMI)
print("Median_BMI:",median_BMI)
print("Mode_BMI:",mode_BMI)

Mean_BMI: 31.992578124999998
Median_BMI: 32.0
Mode_BMI: 0    32.0
Name: BMI, dtype: float64


In [14]:
diabetes.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

He decidido escoger 2 variables:  
Insulin, variable cuantitativa tipo integer  
BMI, variable cuantitativa tipo float

La variable Insulin representa la concentración de insulina en sangre, el rango va desde el valor mínimo de 0 hasta el valor máximo de 846. Un rango así de amplio refleja alta diferencia entre los individuos. Por otro lado, la variable BMI, que quiere decir indice de masa corporal, mide la relación entre el peso y la altura de una persona con un rango desde el minimo 0 hasta el máximo de 67.1. Podemos ver que este rango nos sugiere que la mayoría de las personas están dentro del nivel de obesidad, ya que los valores normales de BMI se encuentran entre 18.5 y 24.9. 

Insulin presenta una media de 79.8 mucho mayor que su mediana de 30.5, esto sumado a la alta desviación estándar de 115.244 indica una distribución asimétrica y muy dispersa. Mientras tanto el BMI tiene media y mediana de 32 y una desviación estándar de 7.88, lo que muestra dentro de lo que cabe una distribución simétrica.

CONSULTAS:

In [18]:
# Personas con niveles altos de insulina
diabetes[diabetes['Insulin'] > 200]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
8              2      197             70             45      543  30.5   
13             1      189             60             23      846  30.1   
16             0      118             84             47      230  45.8   
20             3      126             88             41      235  39.3   
31             3      158             76             36      245  31.6   
..           ...      ...            ...            ...      ...   ...   
707            2      127             46             21      335  34.4   
710            3      158             64             13      387  31.2   
713            0      134             58             20      291  26.4   
715            7      187             50             33      392  33.9   
753            0      181             88             44      510  43.3   

     DiabetesPedigreeFunction  Age  Outcome  
8                       0.158   53        1  
13                      0.398   59        1  
16                      0.551   31        1  
20                      0.704   27        0  
31                      0.851   28        1  
..                        ...  ...      ...  
707                     0.176   22        0  
710                     0.295   24        0  
713                     0.352   21        0  
715                     0.826   34        1  
753                     0.222   26        1  

[85 rows x 9 columns]

In [20]:
len(diabetes[diabetes['Insulin'] > 200])

85

In [19]:
# Personas con BMI bajo
diabetes[diabetes['BMI'] < 18.5]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
9              8      125             96              0        0   0.0   
49             7      105              0              0        0   0.0   
60             2       84              0              0        0   0.0   
81             2       74              0              0        0   0.0   
145            0      102             75             23        0   0.0   
239            0      104             76              0        0  18.4   
371            0      118             64             23       89   0.0   
418            1       83             68              0        0  18.2   
426            0       94              0              0        0   0.0   
438            1       97             70             15        0  18.2   
494            3       80              0              0        0   0.0   
522            6      114              0              0        0   0.0   
526            1       97             64             19       82  18.2   
684            5      136             82              0        0   0.0   
706           10      115              0              0        0   0.0   

     DiabetesPedigreeFunction  Age  Outcome  
9                       0.232   54        1  
49                      0.305   24        0  
60                      0.304   21        0  
81                      0.102   22        0  
145                     0.572   21        0  
239                     0.582   27        0  
371                     1.731   21        0  
418                     0.624   27        0  
426                     0.256   25        0  
438                     0.147   21        0  
494                     0.174   22        0  
522                     0.189   26        0  
526                     0.299   21        0  
684                     0.640   69        0  
706                     0.261   30        1

In [21]:
len(diabetes[diabetes['BMI'] < 18.5])

15

In [22]:
# Personas con rangos normales de BMI e insulina
diabetes[(diabetes['Insulin'] >= 16) & (diabetes['Insulin'] <= 166) & (diabetes['BMI'] >= 18.5) & (diabetes['BMI'] <= 24.9)]

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
27             1       97             66             15      140  23.2   
28            13      145             82             19      110  22.2   
32             3       88             58             11       54  24.8   
50             1      103             80             11       82  19.4   
51             1      101             50             15       36  24.2   
52             5       88             66             21       23  24.4   
68             1       95             66             13       38  19.6   
94             2      142             82             18       64  24.7   
97             1       71             48             18       76  20.4   
119            4       99             76             15       51  23.2   
134            2       96             68             13       49  21.1   
156            2       99             52             15       94  24.6   
197            3      107             62             13       48  22.9   
203            2       99             70             16       44  20.4   
224            1      100             66             15       56  23.6   
252            2       90             80             14       55  24.4   
288            4       96             56             17       49  20.8   
307            0      137             68             14      148  24.8   
316            3       99             80             11       64  19.3   
334            1       95             60             18       58  23.9   
348            3       99             62             19       74  21.8   
384            1      125             70             24      110  24.3   
385            1      119             54             13       50  22.3   
393            4      116             72             12       87  22.1   
396            3       96             56             34      115  24.7   
450            1       82             64             13       95  21.2   
460            9      120             72             22       56  20.8   
465            0      124             56             13      105  21.8   
477            7      114             76             17      110  23.8   
576            6      108             44             20      130  24.0   
599            1      109             38             18      120  23.1   
607            1       92             62             25       41  19.5   
639            1      100             74             12       46  19.5   
646            1      167             74             17      144  23.4   
656            2      101             58             35       90  21.8   
680            2       56             56             28       45  24.2   

     DiabetesPedigreeFunction  Age  Outcome  
27                      0.487   22        0  
28                      0.245   57        0  
32                      0.267   22        0  
50                      0.491   22        0  
51                      0.526   26        0  
52                      0.342   30        0  
68                      0.334   25        0  
94                      0.761   21        0  
97                      0.323   22        0  
119                     0.223   21        0  
134                     0.647   26        0  
156                     0.637   21        0  
197                     0.678   23        1  
203                     0.235   27        0  
224                     0.666   26        0  
252                     0.249   24        0  
288                     0.340   26        0  
307                     0.143   21        0  
316                     0.284   30        0  
334                     0.260   22        0  
348                     0.279   26        0  
384                     0.221   25        0  
385                     0.205   24        0  
393                     0.463   37        0  
396                     0.944   39        0  
450                     0.415   23        0  
460                     

In [23]:
len(diabetes[(diabetes['Insulin'] >= 16) & (diabetes['Insulin'] <= 166) & (diabetes['BMI'] >= 18.5) & (diabetes['BMI'] <= 24.9)])

36